### ISSUE TITLE
ordered-imports does not respect rule flags

### ISSUE THREAD

**Issue by tvald** [palantir/tslint#4372]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.1.4
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
// tslint:disable-next-line:ordered-imports
import './include/polyfills';

import * as React from 'react';
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "ordered-imports": [
      true,
      {
        "grouped-imports": true,
        "import-sources-order": "case-insensitive",
        "named-imports-order": "lowercase-last"
      }
    ]
  }
}
```

#### Actual behavior
Warning on `react` import:
```
[tslint] Import sources of different groups must be sorted by: libraries, parent directories, current directory. [ordered-imports]
```

#### Expected behavior
No warnings. `ordered-imports` should only consider the set of imports where it is enabled.

**Comment by tvald** [comment]
This may just be an unintuitive edge case, rather than an actual bug. Moving the warning to the second import line suppresses the warning (and it does not occur on following import lines either):

```ts
import './include/common';

// tslint:disable-next-line:ordered-imports
import * as React from 'react';
import * as SomethingElse from 'something-else';
```

**Comment by JoshuaKGoldberg** [comment]
For reference, `ordered-imports` has to only give a complaint on the last available line because of #2556. It would be dangerous for it to highlight all the imports instead of the last line because overlapping import conflicts might arise from variable names (e.g. [`tslint-microsoft-contrib`'s `import-name` rule](https://github.com/microsoft/tslint-microsoft-contrib)).

Once overlapping fixes are better handled, this error should go away. Glad you have a workaround in the meantime @tvald! 

Separately, it looks like you have a valid use case for `ordered-imports` that isn't yet handled by the rule's configuration (?). If that's the case, you could file an issue asking for the rule's config to be more extensible.
